### PR TITLE
Implement Visibility property in ViewHandlers

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement `Visibility` property in ViewHandlers.

- Collapsed: Do not display the element, and do not reserve space for it in layout.
- Hidden: Do not display the element, but reserve space for the element in layout.
- Visible: Display the element.

Related with https://github.com/dotnet/maui/pull/707
We need to review which approach fits better.

### Platforms Affected ### 

- Core
- iOS
- macOS
- Android
- Windows

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Targets a single property for a single control (or intertwined few properties)
- [x] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [x] Tags ported renderer methods with [PortHandler]
- [x] Adds an example of the property to the sample project (MainPage)
- [x] Adds the property to the stub class
- [x] Implements basic property tests in DeviceTests

### CODE REVIEW

File: src/Core/src/Platform/iOS/ViewExtensions.cs [hunk 1]
```diff
@@ -11,12 +11,33 @@ public static class ViewExtensions
...
+
+			switch (view.Visibility)
+			{
+				case Visibility.Visible:
+					nativeView.Alpha = (float)opacity;
+					nativeView.Hidden = false;
+					break;
+				case Visibility.Hidden:
+					nativeView.Alpha = 0;
+					nativeView.Hidden = false;
```
Comment (mattleibow) [inline src/Core/src/Platform/iOS/ViewExtensions.cs:32]: How does this work with hit test and accessibility?

**Review by mattleibow — COMMENTED** [review]
Looks like it is good. Just wondering what is happening with regards to user interaction (hit test and tab stop) as well as accessibility. And then what _should_ happen. 

Android seems to have this feature, so does that work differently to when we set alpha to 0?

File: src/Core/src/Platform/Windows/ViewExtensions.cs [hunk 2]
```diff
@@ -10,6 +10,27 @@ public static class ViewExtensions
...
+
+			switch (view.Visibility)
+			{
+				case Visibility.Visible:
+					nativeView.Opacity = opacity;
+					nativeView.Visibility = UI.Xaml.Visibility.Visible;
+					break;
+				case Visibility.Hidden:
+					nativeView.Opacity = 0;
+					nativeView.Visibility = UI.Xaml.Visibility.Visible;
```
[the remark this answers is not in the document]
Comment (hartez) [reply src/Core/src/Platform/Windows/ViewExtensions.cs:26]: We don't know yet how they _should_ work, so I think it's moot for now.

**Review by hartez — APPROVED** [review]

**Review by mattleibow — APPROVED** [review]

**Comment by greatoceansoftware** [comment]
Did you give up on Visibility (Visible, Hidden, Collapsed) in favor of IsVisible? I can see the enum in the Object Browser, but it's not appearing on any controls yet. I hope I'm misinterpreting this thread incorrectly. Really need Visibility implemented.

**Pull request merged**